Server/ReasnAPI/ReasnAPI/Services/UserService.cs

```diff
@@ -1,16 +1,136 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class UserService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+using Microsoft.EntityFrameworkCore;
+using ReasnAPI.Mappers;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using ReasnAPI.Services.Exceptions;
+using System.Linq.Expressions;
+using System.Transactions;
+
+namespace ReasnAPI.Services;
+
+public class UserService(ReasnContext context)
+{
+    public UserDto? UpdateUser(int userId, UserDto? userDto)
+    {
+        using (var scope = new TransactionScope())
+        {
+            ArgumentNullException.ThrowIfNull(userDto);
+
+            var user = context.Users
+                                .Include(u => u.UserInterests)
+                                .ThenInclude(ui => ui.Interest)
+                                .FirstOrDefault(r => r.Id == userId);
+
+            if (user is null)
+            {
+                throw new NotFoundException("User not found");
+            }
+
+            var usernameExists = context.Users.Any(r => r.Username == userDto.Username && r.Id != userId);
+            var emailExists = context.Users.Any(r => r.Email == userDto.Email && r.Id != userId);
+            var phoneExists = context.Users.Any(r => r.Phone == userDto.Phone && r.Id != userId);
+
+            if (usernameExists)
+            {
+                throw new BadRequestException("Username already exists");
+            }
+
+            if (emailExists)
+            {
+                throw new BadRequestException("Email already exists");
+            }
+
+            if (phoneExists)
+            {
+                throw new BadRequestException("Phone already exists");
+            }

```

Comment:
Why send three possible requests to database when it can fail on the first "if statement"
"?